.golangci.yml

```diff
@@ -13,107 +9,12 @@ run:
   concurrency: 6
   deadline: 5m
 linters:
-  enable-all: true
-  disable:
-    - asciicheck
-    - errcheck
-    - exhaustive
-    - funlen
-    - gochecknoglobals
-    - gochecknoinits
-    - gocognit
-    - goconst
-    - godot
-    - godox
-    - goerr113
-    - gofumpt
-    - golint
-    - gomnd
-    - gosec
-    - lll
-    - maligned
-    - nestif
-    - nlreturn
-    - prealloc
-    - stylecheck
-    - testpackage
-    - unconvert
-    - whitespace
-    - wsl
+  enable:
+    - dupl
+    - gocritic
```

Comment:
I personally prefer a disabled list as we can benefit from new linters when updating golangci-lint.  I find it also easier to enable linters as they are explicitly listed.

Reply:
I have said something about this in commit message.

From the maintenance perspective, having a disabled list is not good, meaning one can't just update the version and go ahead, for the exact same reason you mentioned -- new linters. Note that not all of them should be used by default. Some are very opinionated. Some are not always right. Some impose restrictions or coding style we do not use in this project.

Perhaps this was the reason why golangci-lint was not updated in this repo for quite some time (v1.30, currently used, has been released Aug 02, 2020). It will benefit to update golangci-lint more often even without enabling new linters, since the ones we use are also being updated, improved, fixed for recent golang.

My perspective is, we should go with the default list, and enable selected linters beyond the default set.

Reply:
Quoting the commit message from the last commit:
```
1.2. Instead of enabling all the linters and then disabling many of
     them, go with the default set of enabled linters, and then add some
     (currently just gocritic and dupl).

     This will hopefully make upgrading golangci-lint easier, as it
     maintains a sensitive set of linters enabled by default (and
     enabling all linters by default means we will enable some new and
     unknown linters when we bump golangci-lint version).
```

Reply:
Curious what others think. @containers/podman-maintainers 

> Perhaps this was the reason why golangci-lint was not updated in this repo for quite some time (v1.30, currently used, has been released Aug 02, 2020).

I have no data to back it but I don't think that's the reason. c/common has less traffic than other projects and such maintenance tasks can easily be lost on the radar.

> It will benefit to update golangci-lint more often even without enabling new linters, since the ones we use are also being updated, improved, fixed for recent golang.

In my experience, it's not that much work to update golangci-lint and selectively disable new linters that break. As you mentioned, some are very opinionated which we can keep disabled. But others have proven to find quite some code smells (and bugs) in Podman.

In other words, I have made positive experiences with a disabled list.

Reply:
> But others have proven to find quite some code smells (and bugs) in Podman.

Well, if you know any good linters that are not included into the default list, let's enable those! Having said that, I'd rather do that when I want to, on my own time, and not when it's time to update golangci-lint to a newer version.

Here's a practical example: golangci-lint < 1.45 does not play well with Go 1.18. Once we switch to Go 1.18, we have to bump it to 1.45. In case all linters are enabled by default, such bump won't be easy -- we'd have to look into various new linters and their errors. Most probably, we'll just disable them without much looking into because at the moment we're in a rush to fix CI, which is expected to always work.

OTOH, when you have time on e.g. a rainy Friday night, you can take a look at what non-default linters are there, and try using those one by one, maybe finding a few jewels, maybe not.

To sum it up, I think that decoupling golangci-lint update process from the process of enabling new linters is a good thing to have.

Reply:
Another argument is "keep it simple", in this case applicable to golangci-lint configuration file. The bigger it is, the harder it is to maintain (since this is not something we're looking at every day, unlike our codebases). Bigger config means we're farther from the default, well tested, settings. Bigger config means more maintenance down the road.

In particular, this PR changes `.golangci.yml` from 126 lines down to 28 lines. I see it as a win.

Reply:
This argument is getting too expensive for me.  I made it clear _why_ we have been rolling this way.  If you feel _that_ strongly about inverting, I am OK. Let's put you in the OWNERS file so that you take responsibility for maintaining the golangci.yml.

Reply:
-> https://github.com/containers/common/pull/1002